File: it/structures/js/default_naming-default/lower_snake.js
export class Value {
  constructor(foo_bar) {
    this.foo_bar = foo_bar;
  }

  static decode(data) {
    let v_foo_bar = data["foo_bar"];

    if (v_foo_bar === null || v_foo_bar === undefined) {
      throw new Error("foo_bar" + ": required field");
    }

    if (typeof v_foo_bar !== "string") {
      throw Error("expected string");
    }

    return new Value(v_foo_bar);
  }

  encode() {
    const data = {};

    if (this.foo_bar === null || this.foo_bar === undefined) {
      throw new Error("foo_bar: is a required field");
    }

    data["foo_bar"] = this.foo_bar;

    return data;
  }
}

File: it/structures/dart/default_naming-default/lower_snake.dart
class Value {
  String fooBar;

  Value(this.fooBar);

  static Value decode(dynamic data) {
    if (!(data is Map<String, dynamic>)) {
      throw "expected Map<String, dynamic> but got $data";
    }

    Map<String, dynamic> _data = data;

    var fooBar_dyn = _data["foo_bar"];

    if (fooBar_dyn == null) {
      throw "expected value but was null";
    }

    if (!(fooBar_dyn is String)) {
      throw "expected String, but was: fooBar_dyn";
    }
    final String fooBar = fooBar_dyn;

    return Value(fooBar);
  }

  Map<String, dynamic> encode() {
    Map<String, dynamic> _data = Map();

    _data["foo_bar"] = this.fooBar;

    return _data;
  }
}
